src/js/components/Pagination/Pagination.js

```diff
@@ -199,27 +204,69 @@ const Pagination = forwardRef(
       ...navProps[control],
     }));
 
+    const paginationControls = (
+      <Nav
+        a11yTitle={ariaLabel || a11yTitle || 'Pagination Navigation'}
+        ref={ref}
+      >
+        <Box as="ul" {...theme.pagination.controls}>
+          {controls.map((control, index) => (
+            /* Using index as key (as opposed to a unique id) seems to
+             * help React prioritize rendering the updated controls as
+             * desired. Whereas, using a unique id resulted in rendering
+             * the active control with an undesired lag. */
+            // eslint-disable-next-line react/no-array-index-key
+            <PageControl key={index} size={size} {...control} />
+          ))}
+        </Box>
+      </Nav>
+    );
+
+    // for backwards compatibility
+    if (!summary && !stepOptions)
+      return (
+        <StyledPaginationContainer
+          flex={false}
+          {...theme.pagination.container}
+          {...rest}
+        >
+          {paginationControls}
+        </StyledPaginationContainer>
+      );
+
     return (
       <StyledPaginationContainer
+        direction="row"
```

Comment:
Can we try the following?

```
<StyledPaginationContainer
        direction="row"
        align="center"
        gap={{ column: 'medium', row: 'xsmall' }}
        wrap
        flex={false}
        {...theme.pagination.container}
        {...rest}
      >
        <Box direction="row" flex="grow">
          {summary && (
            <PaginationSummary
              messages={messages}
              page={page}
              step={step}
              numberItems={total}
              pad={{ vertical: 'small' }}
            />
          )}
        </Box>
        <Box direction="row" gap={{ column: 'small', row: 'small' }} wrap>
          {stepOptions && (
            <PaginationStep
              messages={messages}
              options={Array.isArray(stepOptions) ? stepOptions : undefined}
              step={step}
              onChange={({ value }) => setStep(value)}
            />
          )}
          {paginationControls}
        </Box>
      </StyledPaginationContainer>
```

Reply:
I think second Box here may need an align="center". More apparent that this is missing when viewing the Grommet theme.